data-prepper-plugins/s3-sink/src/test/java/org/opensearch/dataprepper/plugins/sink/S3SinkTest.java

```diff
@@ -0,0 +1,159 @@
+
+        pluginSetting = mock(PluginSetting.class);
+        pluginModel = mock(PluginModel.class);
+        pluginFactory = mock(PluginFactory.class);
+
+        when(s3SinkConfig.getThresholdOptions()).thenReturn(thresholdOptions);
+        when(s3SinkConfig.getThresholdOptions().getEventCount()).thenReturn(100);
+        when(s3SinkConfig.getThresholdOptions().getMaximumSize()).thenReturn(ByteCount.parse("1kb"));
+        when(s3SinkConfig.getThresholdOptions().getEventCollect()).thenReturn(Duration.ofSeconds(5));
+
+        when(objectKeyOptions.getNamePattern()).thenReturn("my-elb-%{yyyy-MM-dd'T'hh-mm-ss}");
+
+        when(s3SinkConfig.getAwsAuthenticationOptions()).thenReturn(awsAuthenticationOptions);
+        when(awsAuthenticationOptions.getAwsRegion()).thenReturn(Region.of("us-east-1"));
+        when(awsAuthenticationOptions.authenticateAwsConfiguration()).thenReturn(awsCredentialsProvider);
+
+        when(s3SinkConfig.getCodec()).thenReturn(pluginModel);
+        when(pluginModel.getPluginName()).thenReturn("json");
+        when(pluginFactory.loadPlugin(Codec.class, pluginSetting)).thenReturn(codec);
+
+        when(pluginSetting.getName()).thenReturn("s3");
+        when(pluginSetting.getPipelineName()).thenReturn("S3-sink-pipeline");
+    }
+
+    @Test
+    void test_s3_sink_plugin_isReady_true() {
+        s3Sink = new S3Sink(pluginSetting, s3SinkConfig, pluginFactory);
+        s3Sink.doInitialize();
+        assertTrue(s3Sink.isReady(), "s3 sink is not ready to work");
+    }
+
+    @Test
+    void test_s3_Sink_plugin_isReady_false() {
+        s3Sink = new S3Sink(pluginSetting, s3SinkConfig, pluginFactory);
+        assertFalse(s3Sink.isReady(), "s3 sink is ready to work");
+    }
+
+    @Test
+    void test_doOutput_with_exception() {
+        s3Sink = new S3Sink(pluginSetting, s3SinkConfig, pluginFactory);
+        when(s3SinkConfig.getThresholdOptions()).thenReturn(null);
+        assertThrows(NullPointerException.class, s3Sink::doInitialize);
+    }
+
+    @Test
+    void test_doOutput_without_data() {
+        s3Sink = new S3Sink(pluginSetting, s3SinkConfig, pluginFactory);
+        s3Sink.doInitialize();
+        s3Sink.doOutput(new ArrayList<>());
+        assertNotNull(s3Sink);
+    }
+
+    @Test
+    void test_doOutput_with_data_local_file() {
+        when(s3SinkConfig.getBufferType()).thenReturn(BufferTypeOptions.LOCALFILE);
+        s3Sink = new S3Sink(pluginSetting, s3SinkConfig, pluginFactory);
+        s3Sink.doInitialize();
+        s3Sink.doOutput(generateRandomStringEventRecord());
+        assertNotNull(s3Sink);
+    }
+
+    @Test
+    void test_doOutput_with_data_in_memory() {
+        when(s3SinkConfig.getBufferType()).thenReturn(BufferTypeOptions.INMEMORY);
+        s3Sink = new S3Sink(pluginSetting, s3SinkConfig, pluginFactory);
+        s3Sink.doInitialize();
+        s3Sink.doOutput(generateRandomStringEventRecord());
+        assertNotNull(s3Sink);

```

Comment:
Please validate if `doOutput()` worked as expected.

Reply:
Yes, this is very important.

You should be able to perform the following:

```
InOrder inOrder = inOrder(s3SinkService);
inOrder.verify(s3SinkService).processRecords(records);
inOrder.verify(s3SinkService).accumulateBufferEvents(sinkWorker);
```

Reply:
Updated test-cases using inOrder and verified.

Reply:
Have you uploaded new diffs?

Reply:
Resolved.